eyecite/tokenizers.py

```diff
@@ -528,6 +528,12 @@ def convert_regex(regex):
                         )
                     except TypeError:
                         hyperscan_db = hyperscan.loadb(cache_bytes)
+                    except hyperscan.InvalidError:
```

Comment:
Hm, is it possible that some error would be silenced because of this, leading a user to think it's working properly when it's not?

Reply:
> is it possible that some error would be silenced because of this, leading a user to think it's working properly when it's not?

I don't think so, I checked the [python-hyperscan ](https://github.com/darvid/python-hyperscan) code and I noticed the `loadb` function relies on Hyperscan's `hs_deserialize_database` for internal database loading., Here's a fragment of the code: 

```py
hs_error_t err = hs_deserialize_database(buf, length, &db->hs_db);
HANDLE_HYPERSCAN_ERR(err, NULL);
if (PyErr_Occurred())
  return NULL;
return odb;
```

The rest of the code is here: [PyObject *loadb](https://github.com/darvid/python-hyperscan/blob/0103e2a3c4f17eb19b1164b871b31b1d5699f567/src/hyperscan/hyperscanmodule.c#L1247-L1270)


Checking the `hs_deserialize_database` method, we can see that this function returns the error (HS_INVALID) if the sanity checks on the provided serialized data (bytes) or the output database pointer (db) fail. Rebuilding the database guarantees a reliable and usable object for the code's future operations.

Here's a fragment of the [hyperscan method](https://github.com/intel/hyperscan/blob/bc3b191ab56055e8560c7cdc161c289c4d76e3d2/src/database.c#L243-L290):

```c
hs_error_t HS_CDECL hs_deserialize_database(const char *bytes,
                                            const size_t length,
                                            hs_database_t **db) {
    if (!bytes || !db) {
        return HS_INVALID;
    }

    // Decode and check the header
    // The db_decode_header method could also returns HS_INVALID
    hs_database_t header;
    hs_error_t ret = db_decode_header(&bytes, length, &header);
    if (ret != HS_SUCCESS) {
        return ret;
    }

   ... 

   if (db_check_crc(tempdb) != HS_SUCCESS) {
        hs_database_free(tempdb);
        return HS_INVALID;
    }

    *db = tempdb;
    return HS_SUCCESS;
}
```

While the same exception can occur with other Hyperscan methods (like `scan`), this try-except clause only handles the exception when the loadb function raises the error.